docs/guides/end-to-end-testing/testing-strategies/azure-active-directory-authentication.mdx

```diff
@@ -0,0 +1,242 @@
+
+When finished, your SPA should be running on `http://localhost:3000` and be
+properly configured to run with Azure Active Directory.
+
+## Configuring Cypress to use Microsoft AAD
+
+To have access to test user credentials within our tests, we need to configure
+Cypress to use the
+[Azure Active Directory](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-whatis)
+user credentials for a given user in your tenant. These user credentials should
+be set in the `cypress.env.json` file inside the root directory of your project.
+
+```json
+// cypress.env.json
+{
+  "aad_username": "AAD_USERNAME",
+  "aad_password": "AAD_PASSWORD",
+  "aad_name": "AAD_NAME"
+}
+```
+
+Also, to authenticate with Microsoft or Azure Active Directory, you will need to
+enable the
+[`experimentalModifyObstructiveThirdPartyCode`](/guides/guides/web-security#Modifying-Obstructive-Third-Party-Code)
+configuration option. If not enabled, your authentication workflow will enter an
+infinite redirect loop.
+
+:::cypress-config-example
+
+```js
+{
+  e2e: {
+    experimentalModifyObstructiveThirdPartyCode: true
+  }
+}
+```
+
+:::
+
+## Login with [`cy.origin()`](/api/commands/origin)
+
+Next, we'll write a custom command called `loginToAAD` to perform a login to
+[Azure Active Directory](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-whatis).
+This command will use [`cy.origin()`](/api/commands/origin) to
+
+1. Navigate to the Azure Active Directory login page on
+   `login.microsoftonline.com`
+2. Input user credentials
+3. Sign in and redirect back to the demo application
+4. Cache the results with [`cy.session()`](/api/commands/session)
+
+```js
+// cypress/support/e2e.ts
+
+function loginViaAAD(username: string, password: string) {
+  cy.visit('http://localhost:3000/')
+  cy.get('button#signIn').click()
+
+  // Login to your AAD tenant.
+  cy.origin(
+    'login.microsoftonline.com',
+    {
+      args: {
+        username,
+      },
+    },
+    ({ username }) => {
+      cy.get('input[type="email"]').type(username, {
+        log: false,
+      })
+      cy.get('input[type="submit"]').click()
+    }
+  )
+
+  // depending on the user and how they are registered with Microsoft, the origin may go to live.com
+  cy.origin(
+    'login.live.com',
+    {
+      args: {
+        password,
+      },
+    },
+    ({ password }) => {
+      cy.get('input[type="password"]').type(password, {
+        log: false,
+      })
+      cy.get('input[type="submit"]').click()
+      cy.get('#idBtn_Back').click()
+    }
+  )
+
+  // Ensure Microsoft has redirected us back to the sample app with our logged in user.
+  cy.url().should('equal', 'http://localhost:3000/')
+  cy.get('#welcome-div').should(
+    'contain',
+    `Welcome ${Cypress.env('aad_username')}!`
+  )
+}
+
+Cypress.Commands.add('loginToAAD', (username: string, password: string) => {
+  const log = Cypress.log({
```

Comment:
We really should not be sharing these examples in Guides IMO because it requires the user to clean these up themselves if a failure occurs. Typically these will be left as pending when a failure occurs.

Reply:
Do you mean the added log message? I am not a big fan of it either for that, but mostly because it just adds code that really isn't necessary for understanding how this needs to work. The only reason I keep it here is to be consistent with the other guides. I wonder if we should do a pass at removing that in the docs, but keep it in the RWA if applicable?

Reply:
I'll likely keep it in, but I don't like the pattern in our guides. I think in the future we should remove it from the others as well, even programmatic login